packages/openactive-broker-microservice/app.js

```diff
@@ -91,28 +98,36 @@ function getRPDE(url, headers, cb) {
         throw `Base URL of RPDE 'next' property ("${getBaseUrl(json.next)}") does not match base URL of RPDE page ("${url}")`;
       }
 
+      context.currentPage = url;
       if (json.next == url && json.items.length == 0) {
         if (WAIT_FOR_HARVEST) {
           feedUpToDate(url);
         } else {
-          console.log(`Sleep mode poll for RPDE feed "${url}"`);
+          if (VERBOSE) console.log(`Sleep mode poll for RPDE feed "${url}"`);
         }
-        setTimeout(x => getRPDE(url, headers, cb), 500);
+        context.sleepMode = true;
+        if (context.timeToHarvestCompletion === undefined) context.timeToHarvestCompletion = millisToMinutesAndSeconds(new Date() - startTime);
+        setTimeout(x => getRPDE(url, context, cb), 500);
       } else {
-        cb(json, headers);
+        context.responseTimes.push(response.elapsedTime);
+        if (context.responseTimes.length > 5) context.responseTimes.shift();
```

Comment:
Comment to explain that you're maintaining a "last 5 items" buffer?

Reply:
Great point, will add